src/memex/presenters.py

```diff
@@ -6,10 +6,13 @@
 import collections
 import copy
 
+from pyramid import security
+
 
 class AnnotationBasePresenter(object):
-    def __init__(self, annotation, links_service):
-        self.annotation = annotation
+    def __init__(self, annotation_resource, links_service):
```

Comment:
Isn't it much nicer to change the presenter to always use the resource rather than the model? This way the presenter does not need to know anything about the group service, it just uses the ACL of the passed-in resource.